src/Components/MainArea/Bookshelf/Bookshelf.tsx

```diff
@@ -1,5 +1,19 @@
 import React from 'react';
+import { IBooks } from '../../../typings/IBooks';
+import { Bookcase } from './Bookcase/Bookcase';
 
-const Bookshelf = () => <div>Bookshelf</div>;
+interface IProps {
+    books: IBooks[];
+}
+
+const Bookshelf = ({ books }: IProps) => {
```

Comment:
Destructuring is awesome in functions, but I found it to be not scaleable in function components. When you have visual only component very soon you will want to customize it with sending DOM properties using spread operator like that - `<div {...props}>` therefore I always prefer to pass props to the function component and then split it to props and domProps like this - `const { books, ...domProps} = props`

Reply:
You right on this one. It looks fancy but could make problems in the future.